src/librustc_typeck/coherence/overlap.rs

```diff
@@ -65,6 +65,23 @@ impl<'cx, 'tcx> OverlapChecker<'cx, 'tcx> {
                 continue;
             }
 
+            if self.tcx.default_trait_impls.borrow().contains(&trait_def_id) {
```

Comment:
I'll make `default_trait` private in `ty`. I've a branch where I'm doing this for other maps/sets in `ty`.